feat(zombie): change zombie abilities

kemperhun opened the pull request:
Способность зомби Consume теперь заражает жертву и восстанавливает голод зомби. 
Способность зомби Breath Death усиливает голод и не может быть использована, когда зомби слишком голоден. А еще увеличен ее делей.

<details>
<summary>Чейнджлог</summary>

```yml
🆑
bugfix: Пожирание зомби теперь восстанавливает голод
balance: Увеличен делей у дыхания зомби. Теперь его нельзя использовать, если зомби слишком голоден
/🆑
```

</details>

- [x] Pull Request полностью завершен, мне не нужна помощь чтобы его закончить.
- [x] Я внимательно прочитал все свои изменения и багов в них не нашел.
- [x] Я запускал сервер со своими изменениями локально и все протестировал.
- [x] Я ознакомился c [Guide to Contribute](https://github.com/ChaoticOnyx/OnyxBay/blob/dev/docs/contributing.md).

kemperhun committed:
feat(zombie): change zombie abilities

Способность зомби Consume теперь заражает жертву и восстанавливает голод зомби.
Способность зомби Breath Death усиливает голод и не может быть использована, когда зомби слишком голоден. А еще увеличен ее делей.

ArmenianOnyx commented:
И что мы имеем: 
Зомби может использовать абилку заражения всего 3 раза, с периодичностью раз в 10 секунд.
(src.nutrition < 150)
src.nutrition -= 50

А потом ему придется медленно ползти с пустым желудком до ближайшего трупа, что бы жрать труп и восстанавливать голод в 2,5 раза медленнее.
src.nutrition += 20

Вы уже пилите нерф говна для педального режима, которого собственно и нету?

ghost commented:
С учетом нашего онлайна это более-менее приемлемые статы, Армян.

ArmenianOnyx commented:
Ивентовый режим не рассчитан на мизерный онлайн. На то он и ивентовый.

> С учетом нашего онлайна это более-менее приемлемые статы, Армян.

kemperhun commented:
Армян не умеет читать код. nutrition = 150 - значение старвинга космонавтика (когда бургер с красной стрелочкой появляется), это НИЖНЯЯ граница, а не верхняя. 

При большом желании можно сожрать всю кухню и с овервейтом колобком кататься и заражать всех подряд.

kemperhun committed:
NoTips loh

TobyThorne commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -474,3 +474,66 @@
+		return
+
+	last_special = world.time + 10 SECONDS
+	src.nutrition -= 50
```
```suggestion
	nutrition -= 50
```

TobyThorne commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -474,3 +474,66 @@
+		to_chat(src, SPAN("warning", "You can't do that while you're incapacitated!"))
+		return
+	if (src.nutrition < 150)
+		to_chat(src, SPAN_WARNING("You are too hungry to spread toxins!"))
```
```suggestion
		to_chat(src, SPAN("warning", "You are too hungry to spread toxins!"))
```

TobyThorne commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -474,3 +474,66 @@
+	if (incapacitated(INCAPACITATION_DISABLED))
+		to_chat(src, SPAN("warning", "You can't do that while you're incapacitated!"))
+		return
+	if (src.nutrition < 150)
```
```suggestion
	if (nutrition < 150)
```

TobyThorne commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -474,3 +474,66 @@
+	set category = "Abilities"
+
+	if (last_special > world.time)
+		to_chat(src, SPAN_WARNING("You aren't ready to do that! Wait [round(last_special - world.time) / 10] seconds."))
```
```suggestion
		to_chat(src, SPAN("warning", "You aren't ready to do that! Wait [round(last_special - world.time) / 10] seconds."))
```

TobyThorne commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -474,3 +474,66 @@
+		return
+
+	if (incapacitated())
+		to_chat(src, SPAN_WARNING("You can't do that while you're incapacitated!"))
```
```suggestion
		to_chat(src, SPAN("warning", "You can't do that while you're incapacitated!"))
```

TobyThorne commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -474,3 +474,66 @@
+			target = L
+			break
+	if (!target)
+		to_chat(src, SPAN_WARNING("You aren't on top of a victim!"))
```
```suggestion
		to_chat(src, SPAN("warning", "You aren't on top of a victim!"))
```

TobyThorne commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -474,3 +474,66 @@
+
+	last_special = world.time + 5 SECONDS
+
+	src.visible_message(SPAN_DANGER("\The [src] hunkers down over \the [target], tearing into their flesh."))
```
```suggestion
	visible_message(SPAN("danger", "\The [src] hunkers down over \the [target], tearing into their flesh."))
```

TobyThorne commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -474,3 +474,66 @@
+
+	src.visible_message(SPAN_DANGER("\The [src] hunkers down over \the [target], tearing into their flesh."))
+	if(do_mob(src, target, 5 SECONDS))
+		to_chat(target, SPAN_DANGER("\The [src] scrapes your flesh from your bones!"))
```
```suggestion
		to_chat(target, SPAN("danger", "\The [src] scrapes your flesh from your bones!"))
```

TobyThorne commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -474,3 +474,66 @@
+	src.visible_message(SPAN_DANGER("\The [src] hunkers down over \the [target], tearing into their flesh."))
+	if(do_mob(src, target, 5 SECONDS))
+		to_chat(target, SPAN_DANGER("\The [src] scrapes your flesh from your bones!"))
+		to_chat(src, SPAN_DANGER("You feed hungrily off \the [target]'s flesh."))
```
```suggestion
		to_chat(src, SPAN("danger", "You feed hungrily off \the [target]'s flesh."))
```

TobyThorne commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -474,3 +474,66 @@
+				infect_virus2(target, D)
+		if(target.getBruteLoss() < -target.maxHealth)
+			target.gib()
+		src.adjustBruteLoss(-25)
```
```suggestion
		adjustBruteLoss(-25)
```

TobyThorne commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -474,3 +474,66 @@
+			for(var/ID in src.virus2)
+				var/datum/disease2/disease/D = src.virus2[ID]
+				infect_virus2(target, D)
+		if(target.getBruteLoss() < -target.maxHealth)
```
это вообще чё за рофл

TobyThorne replied:
```suggestion
		if(target.getBruteLoss() > target.maxHealth)
```

TobyThorne commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -474,3 +474,66 @@
+		if(target.getBruteLoss() < -target.maxHealth)
+			target.gib()
+		src.adjustBruteLoss(-25)
+		src.nutrition += 20
```
```suggestion
		nutrition += 20
```

kemperhun committed:
спаны и срц

Если вы меняете все вхождения слова "форон" на слово "плазма" - неужели так сложно заменить все спаны в билде тем же способом?

kemperhun committed:
Merge remote-tracking branch 'upstream/dev' into Change-zombie-powers

TobyThorne reviewed (approved):
достер тестил инфа соточка
